tests/components/snmp/test_integer_switch.py

```diff
@@ -0,0 +1,84 @@
+"""SNMP switch tests."""
+
+from unittest.mock import patch
+
+from pysnmp.hlapi import Integer32
+
+from homeassistant.components.switch import DOMAIN as SWITCH_DOMAIN
+from homeassistant.core import HomeAssistant
+from homeassistant.setup import async_setup_component
+
+
+async def test_snmp_integer_switch_off(hass: HomeAssistant) -> None:
+    """Test snmp switch returning int 0 for off."""
+
+    config = {
+        SWITCH_DOMAIN: {
+            "platform": "snmp",
+            "host": "192.168.1.32",
+            # ippower-mib::ippoweroutlet1.0
+            "baseoid": "1.3.6.1.4.1.38107.1.3.1.0",
+            "payload_on": 1,
+            "payload_off": 0,
+        },
+    }
```

Comment:
Rather than repeating the config 3 times, should we make it a constant that is referenced.

Reply:
Will do, thanks!

Reply:
Done in 550649a0c7315d5d8ae250c1b04c4e13ff48a35c